CIPs/cip-127.md

```diff
@@ -0,0 +1,207 @@
+---
+cip: 127
+title: CapReg - object-capability registry
+author: Joel Thorstensson (@oed)
+discussions-to: https://forum.ceramic.network/t/cip-127-capreg-object-capability-registry
+status: Draft
+category: RFC
+created: 2023-04-28
+updated: 2023-04-28
+---
+
+## Simple Summary
+
+<!--Provide a simplified and layman-accessible explanation of the CIP.-->
+The CapReg capability registry enables users to notarize and revoke any object-apability associated with their DID.
+
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+Using a simple Ceramic stream, the capability registry enables users to notarize and revoke object-capabilities issued by their DID. The registry is based on hashes of object-capabilities encoded as CACAO.
+
+
+## Motivation
+
+<!--Motivation is critical for CIPs that want to change the Ceramic protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.-->
+
+Currently in Ceramic the main accounts types are PKH DID. These are great because they enable existing wallets to be used directly with Ceramic. Once a PKH DID is used to delegate permissions to a session key. That delegation will remain valid until the capability expires. This could be a problem in the case of a stolen session key or a malicious application. By introducing a capability registry these capabilities can be revoked at any time by the main DID (PKH DID in the case above, but this would work for any DID method). A system that uses these object capabilities could refer to the registry to verify that the capability  has not been revoked before it was used.
```

Comment:
```suggestion
Currently in Ceramic the main accounts types are PKH DID. These are great because they enable existing wallets to be used directly with Ceramic. Once a PKH DID is used to delegate permissions to a session key, that delegation will remain valid until the capability expires. This could be a problem in the case of a stolen session key or a malicious application. By introducing a capability registry these capabilities can be revoked at any time by the main DID (PKH DID in the case above, but this would work for any DID method). A system that uses these object capabilities could refer to the registry to verify that the capability  has not been revoked before it was used.
```